library/general/block-wildcard-ingress/suite.yaml

```diff
@@ -0,0 +1,17 @@
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+metadata:
+  name: block-wildcard-ingress
+tests:
+- name: block-wildcard-ingress
+  template: template.yaml
+  constraint: samples/block-wildcard-ingress/constraint.yaml
+  cases:
+  - name: example-allowed
+    object: samples/block-wildcard-ingress/example_allowed.yaml
+    assertions:
+    - violations: 0
+  - name: example-disallowed
```

Comment:
Rather than a single `example_disallowed.yaml`, you can separate the negative test cases into three separate objects.  Each object can have a descriptive name like `host-name-omitted.yaml`, and could even be in a folder like `library/general/block-wildcard-ingress/samples/block-wildcard-ingress/disallowed/host-name-omitted.yaml`.

This would mean that if one of the edge cases you test was broken by a future change to the template, a test for that specific edge case would fail. It's also easier for users to understand the specific cases that are disallowed, since they are spelled out explicitly in the tests.  I had to go read your rego to understand what about each ingress rule was being tested.